Userland/Libraries/LibWeb/Loader/ResourceLoader.cpp

```diff
@@ -250,6 +255,61 @@ void ResourceLoader::load(LoadRequest& request, Function<void(ReadonlyBytes, Has
 
             auto const fd = file_or_error.value();
 
+            // When local file is a directory generate directory entries page
+            auto maybe_is_valid_directory = Core::Directory::is_valid_directory(fd);
+            if (!maybe_is_valid_directory.is_error() && maybe_is_valid_directory.value()) {
+                // Generate HTML contents entries list
+                auto lexical_path = LexicalPath(request.url().serialize_path());
+                Core::DirIterator dt(lexical_path.string(), Core::DirIterator::Flags::SkipParentAndBaseDir);
+                Vector<DeprecatedString> names;
+                while (dt.has_next())
+                    names.append(dt.next_path());
```

Comment:
You could use `Core::Directory::for_each_entry()` here to gather a list of DirectoryEntries, which know if they are directories or not. Then you wouldn't need the `stat()` call later.